dlp/redact.js

```diff
@@ -64,6 +64,61 @@ function redactString (string, replaceString, minLikelihood, infoTypes) {
   // [END redact_string]
 }
 
+function redactImage (filepath, minLikelihood, infoTypes, outputPath) {
+  // [START redact_image]
+  // Imports required Node.js libraries
+  const mime = require('mime');
+  const fs = require('fs');
+
+  // Imports the Google Cloud Data Loss Prevention library
+  const DLP = require('@google-cloud/dlp');
+
+  // Instantiates a client
+  const dlp = DLP();
+
+  // The path to a local file to inspect. Can be a JPG or PNG image file.
+  // const fileName = 'path/to/image.png';
+
+  // The minimum likelihood required before redacting a match
+  // const minLikelihood = LIKELIHOOD_UNSPECIFIED;
+
+  // The infoTypes of information to redact
+  // const infoTypes = ['EMAIL_ADDRESS', 'PHONE_NUMBER'];
+
+  // The local path to save the resulting image to.
+  // const outputPath = 'result.png';
+
+  const fileItems = [{
+    type: mime.lookup(filepath) || 'application/octet-stream',
+    data: Buffer.from(fs.readFileSync(filepath)).toString('base64')
+  }];
+
+  const imageRedactionConfigs = infoTypes.map((infoType) => {
+    return { infoType: infoType };
+  });
+
+  const request = {
+    inspectConfig: {
+      minLikelihood: minLikelihood
+    },
+    imageRedactionConfigs: imageRedactionConfigs,
+    items: fileItems
+  };
+
+  dlp.redactContent(request)
+    .then((response) => {
+      const image = response[0].items[0].data;
+      fs.writeFileSync(outputPath, image);
```

Comment:
Agreed all around - I was going with the simpler-sample approach here.

We could use [this module](https://www.npmjs.com/package/mz), if we don't mind adding additional dependencies.

Otherwise, I'd personally vote for using the `sync` functions.